addon/components/frost-numeric-select.js

```diff
@@ -0,0 +1,183 @@
+    tabindex: PropTypes.number,
+    type: PropTypes.string,
+    isArrowVisible: PropTypes.bool,
+    isAllowInputNegative: PropTypes.bool,
+    isAllowUseDownButtonToNegative: PropTypes.bool,
+    showError: PropTypes.bool,
+    errorMessage: PropTypes.string,
+    class: PropTypes.string,
+
+    // Setting these as part of establishing an inital value
+    autocapitalize: PropTypes.string,
+    autofocus: PropTypes.bool,
+    autocorrect: PropTypes.string,
+    form: PropTypes.string,
+    maxlength: PropTypes.number,
+    placeholder: PropTypes.string,
+    required: PropTypes.bool,
+    selectionDirection: PropTypes.string,
+    spellcheck: PropTypes.bool,
+    value: PropTypes.string,
+    title: PropTypes.string
+
+    // state
+  },
+
+  getDefaultProps () {
+    return {
+      // options
+      align: 'left',
+      autocapitalize: 'off',
+      autocorrect: 'off',
+      autofocus: false,
+      isArrowVisible: false,
+      isAllowInputNegative: true,
+      isAllowUseDownButtonToNegative:false,
+      showError: false,
+      errorMessage: 'Cannot input invalid character',
+      isHookEmbedded: false,
+      required: false,
+      spellcheck: false,
+      tabindex: 0,
+      type: 'text'
+    }
+  },
+
+  // == Functions =============================================================
+
+  _isNumberKey (e) {
+    let decimalExist = this.get('value') && this.get('value').indexOf('.') >=0
+    let charCode = (e.which) ? e.which : e.keyCode
+
+    if(!this.get('isAllowInputNegative')){
+      if (charCode != 46 && charCode > 31
+        && (charCode < 48 || charCode > 57)) {
+          return false
+       } else if (charCode == 46 && decimalExist) {//check if user can enter decimal point
+         return false
+       }
+      return true
+    } else {
+      if (charCode == 45 && !this.get('value')) {//check if user can enter negative symbol
+        return true
+      } else if (charCode == 46 && decimalExist) {
+        return false
+      } else if (charCode != 46 && charCode > 31
+        && (charCode < 48 || charCode > 57))
+         return false
+
+      return true
+    }
+  },
+
+  //calculate how many digit a value has
+  _calculateDigit (value) {
+    return value.slice(value.indexOf('.')+1).length
+  },
+
+  //set calculated and properly formated value
+  _setCalculatedValue (value, numberOfDigit) {
+    if(this.$('input').val().indexOf('.') > -1) {
+      this.set('value', value.toFixed(numberOfDigit).toString())
+    } else {
+      this.set('value', value.toString())
+    }
+  },
+
+  // == Computed Properties ===================================================
+  /**
+   * Determine which scss class should be used
+   * @param {boolean} isArrowVisible - if the user allow displaying the up and down arrow buttons
+   * @returns {string} - a scss class name
+   */
+  @computed('isArrowVisible')
+  class (isArrowVisible) {
+    return isArrowVisible ? 'frost-numeric-select-button' : 'frost-numeric-select'
```

Comment:
I'd like this to be a conditional modifier on the base class, so something like `.frost-numeric-select` as the base class and `frost-numeric-select.arrow-visible` as the modifier, where the `arrow-visible` would provide the right padding, etc.